src/path.ts

```diff
@@ -460,12 +419,12 @@ class FieldPath extends Path {
    *   });
    * });
    */
-  constructor(segments) {
+  constructor(...segments: string[]) {
     validate.minNumberOfArguments('FieldPath', arguments, 1);
 
-    segments = is.array(segments)
-      ? segments
-      : Array.prototype.slice.call(arguments);
+    segments = segments.length === 1 && is.array(segments[0]) ?
```

Comment:
This change looks semver major. Previously `FieldPath([], 42)` would behave differently than what it does now.

Not sure if this is user-exposed though. If this internal then 👍.